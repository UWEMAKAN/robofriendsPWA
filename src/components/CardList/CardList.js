import React from 'react';

import classes from './CardList.module.css';
import Card from '../Card/Card';

const cardList = ({ robots }) => {
  const cardsArray = robots.map((robot, index) => {
    return <Card key={ index + 1 } name={ robot.name } email={ robot.email } id={ index + 1 } />
  });
  return (
    <div className={ classes.CardList }>
      { cardsArray }
    </div>
  );
}

export default cardList;