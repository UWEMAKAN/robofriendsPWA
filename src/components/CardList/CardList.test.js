import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

import {createSerializer} from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

const mockRobots = [
  {
    id: 1,
    name: 'John Snow',
    email: 'johnsnow@got.com'
  }
];

describe('CardList Component', () => {
  it('it should render CardList component', () => {
    expect(shallow(<CardList robots={ mockRobots } />)).toMatchSnapshot();
  });
});
