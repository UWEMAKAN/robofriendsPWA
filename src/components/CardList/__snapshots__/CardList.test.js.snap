// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component it should render CardList component 1`] = `
<div
  className="CardList"
>
  <card
    email="johnsnow@got.com"
    id={1}
    key="1"
    name="John Snow"
  />
</div>
`;
